### PR TITLE
chore(server): redis error handling

### PR DESCRIPTION
**Pull request by samip5**
Should prevent errors like:

```
missing 'error' handler on this Redis client
missing 'error' handler on this Redis client
```

### CODE REVIEW

File: server/libs/infra/src/infra.config.ts [hunk 1]
```diff
@@ -26,17 +26,24 @@ function parseRedisConfig(): RedisOptions {
...
+export const bullConfig: CustomBullModuleOptions = {
   prefix: 'immich_bull',
   redis: redisConfig,
   defaultJobOptions: {
     attempts: 3,
     removeOnComplete: true,
     removeOnFail: false,
   },
+  onError: (error) => {
+    console.log(error);
```
Comment (bo0tzz) [inline server/libs/infra/src/infra.config.ts:42]: Can this use the nestjs logger instead of console.log?

Comment (jrasm91) [reply server/libs/infra/src/infra.config.ts:42]: Should be able to instantiate a new logger in this life and then use it here.

Comment (samip5) [reply server/libs/infra/src/infra.config.ts:42]: That's a good question as it's not inside a class so one cannot simply do this.logger, but should be possible.

**Review by bo0tzz — APPROVED** [review]

File: server/libs/infra/src/infra.config.ts [hunk 2]
```diff
@@ -26,17 +29,24 @@ function parseRedisConfig(): RedisOptions {
...
     removeOnComplete: true,
     removeOnFail: false,
   },
+  onError: (error) => {
+    logger.error(error);
+  },
 };
 
-export const bullQueues: BullModuleOptions[] = Object.values(QueueName).map((name) => ({ name }));
+export const bullQueues: CustomBullModuleOptions[] = Object.values(QueueName).map((name) => ({ name }));
```
Comment (jrasm91) [inline server/libs/infra/src/infra.config.ts:49]: Is this change necessary?

Comment (samip5) [reply server/libs/infra/src/infra.config.ts:49]: I believe so as the BullModuleOptions doesn't expose the onError, but I could be mistaken too.

Comment (jrasm91) [reply server/libs/infra/src/infra.config.ts:49]: This is only the config for the queues for not main config.

Comment (samip5) [reply server/libs/infra/src/infra.config.ts:49]: Let me change it back as the main config should be enough.

File: server/libs/infra/src/infra.config.ts [hunk 3]
```diff
@@ -26,17 +29,24 @@ function parseRedisConfig(): RedisOptions {
 
 export const redisConfig: RedisOptions = parseRedisConfig();
 
-export const bullConfig: BullModuleOptions = {
+interface CustomBullModuleOptions extends BullModuleOptions {
+  onError?: (error: any) => void;
```
Comment (jrasm91) [inline server/libs/infra/src/infra.config.ts:33]: If there is no type here, how did you figure it this gets called anywhere?

Comment (samip5) [reply server/libs/infra/src/infra.config.ts]: It seems the guidance on stackoverflow wasn't as great. :)

File: server/libs/infra/src/infra.config.ts [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
+
 function parseRedisConfig(): RedisOptions {
   const redisUrl = process.env.REDIS_URL;
   if (redisUrl && redisUrl.startsWith('ioredis://')) {
     try {
       const decodedString = Buffer.from(redisUrl.slice(10), 'base64').toString();
       return JSON.parse(decodedString);
     } catch (error) {
-      throw new Error(`Failed to decode redis options: ${error}`);
+      logger.error(`Failed to decode redis options: ${error}`);
```
Comment (jrasm91) [inline server/libs/infra/src/infra.config.ts:17]: This one should probably throw an error to prevent the app from starting if the config is invalid...

File: server/libs/infra/src/infra.config.ts [hunk 5]
```diff
@@ -54,7 +57,7 @@ function parseTypeSenseConfig(): ConfigurationOptions {
         ...common,
       };
     } catch (error) {
-      throw new Error(`Failed to decode typesense options: ${error}`);
+      logger.error(`Failed to decode typesense options: ${error}`);
```
Comment (jrasm91) [inline server/libs/infra/src/infra.config.ts:60]: Same here

**Review by jrasm91 — APPROVED** [review]
